configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/ObjectMapper.java

```diff
@@ -132,6 +132,50 @@ public void serializeTo(Object instance, ConfigurationNode node) throws ObjectMa
         }
     }
 
+    /* Field (de)serialization if @Adapter annotation is present*/
+    protected static class AdaptedFieldInfo extends FieldData {
+        private FieldAdapter fieldAdapter;
+
+        public AdaptedFieldInfo(Field field, String comment, Class<? extends FieldAdapter> fieldAdapter) throws ObjectMappingException {
+            super(field, comment);
+            try {
+                this.fieldAdapter = fieldAdapter.getDeclaredConstructor().newInstance();
+            } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                throw new ObjectMappingException("Unable to initialize field adapter " + fieldAdapter.getClass().getName());
+            }
+        }
+
+        public void deserializeFrom(Object instance, ConfigurationNode node) throws ObjectMappingException {
+            try {
+                Object newVal = node.isVirtual() ? null : fieldAdapter.deserialize(node);
+                field.set(instance, newVal);
+            } catch (IllegalAccessException e) {
```

Comment:
You should catch a `ClassCastException` and wrap it up here - this might happen if the wrong serialiser is used in the adapater annotation.

Reply:
see a1798168eb0801e45cf8ca47bec282a51a9e490c, no longer relevant